components/squarespace/actions/create-product/create-product.mjs

```diff
@@ -0,0 +1,74 @@
+import squarespace from "../../squarespace.app.mjs";
+import utils from "../common/utils.mjs";
+
+export default {
+  key: "squarespace-create-product",
+  name: "Create Product",
+  description: "Create a new product. [See docs here](https://developers.squarespace.com/commerce-apis/create-product)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    squarespace,
+    storePageId: {
+      propDefinition: [
+        squarespace,
+        "storePageId",
+      ],
+    },
+    variants: {
+      label: "Variants",
+      description: "The variants of the product. Array must contain at least one object but no more than 100 objects. E.g `[ { \"sku\": \"SQ0557856\", \"pricing\": { \"basePrice\": { \"currency\": \"USD\", \"value\": \"12.95\" }, \"onSale\": false, \"salePrice\": { \"currency\": \"USD\", \"value\": \"0.00\" } }, \"stock\": { \"quantity\": 10, \"unlimited\": false }, \"attributes\": { \"Flavor\": \"Habanero\" }, \"shippingMeasurements\": { \"weight\": { \"unit\": \"POUND\", \"value\": 2.0 }, \"dimensions\": { \"unit\": \"INCH\", \"length\": 7.0, \"width\": 5.0, \"height\": 5.0 } } } ]`",
+      type: "string",
```

Comment:
```suggestion
      type: "string[]",
```
Optionally how about a string array here?

Reply:
I chose to use `string` instead of `object` or `string[]` because this types are hard to parse and validate.